### PR TITLE
Feat/allow ens addresses for address input

### LINKED ISSUE #337: Allow input of ENS names
**User Story**
As a proposal creator, I want to input ens into address fields and have the name accepted.

### PR DESCRIPTION
**Pull request by dcrescimbeni**
# Description

- The user can now input an ENS name in any `AddressInput` component.
- The ".eth" at the end is not necessary to correctly parse the address.
- The component returns the address corresponding to that ENS, so the logic and validation of the components using `AddressInput` is unaffected.
- If the ENS name doesn't exist, it returns a null address.
- The component allows disabling en ENS name functionality, to force the user to only input a valid address.
- If the initial value of the component is an address (like when editing an action), it'll try to fetch its corresponding ENS name and display it if available.

Closes #337 

It displays the ENS avatar with a valid ENS name:
Screenshot from 2022-10-04 09-06-47

Invalid ENS name:
Screenshot from 2022-10-04 09-07-12

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Added unit testing and manual testing.

To reproduce:

1. Go to a new proposal page, and create a new action that uses the `AddressInput` component (like "transfer" or "set permissions")
2. Input a valid ENS name in the "Address" field. Wait a few seconds for it to fetch the corresponding address
3. If a valid ENS name is provided, it should show the address avatar
4. If the ENS doesn't exist, it won't show the address, and shouldn't let you submit the form
5. If you input an address that has a corresponding ENS and save the action, upon editing that same action, the ENS name will be shown

# Checklist:

- [x] My code follows the existing style of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any UI changes have been tested and made responsive for mobile views

### CODE REVIEW

**Comment by Kenny-Gin1** [comment]
Yo @dcrescimbeni! Do you think it makes sense to normalize the string? Basically, so that you can enter both `rossdev.eth` and only `rossdev` and both will be hashed the same way.

**Comment by dcrescimbeni** [comment]
> Yo @dcrescimbeni! Do you think it makes sense to normalize the string? Basically, so that you can enter both `rossdev.eth` and only `rossdev` and both will be hashed the same way.

Yeah totally! That should be an easy add.

File: src/components/primitives/Forms/AddressInput/AddressInput.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+  const [localValue, setLocalValue] = useState(
+    (enableENSName && parsedName) || value
+  );
+  const { address: addressFromLocalValue } = useENS(localValue);
+
+  useEffect(() => {
+    if (enableENSName && addressFromLocalValue) onChange(addressFromLocalValue);
+    else onChange(localValue);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (Kenny-Gin1) [inline src/components/primitives/Forms/AddressInput/AddressInput.tsx:32]: I guess this is fine, but we could also do: 

const `_onChange = useCallback(() => onChange(localValue))`. This way we don't need to `eslint-disable` and risk that something is not rendering. But just a minor thing :)

**Review by Kenny-Gin1 — COMMENTED** [review]
Great job ⭐! Just a few comments

File: src/components/primitives/Forms/AddressInput/AddressInput.tsx (hunk 1 above)
Comment (dcrescimbeni) [reply src/components/primitives/Forms/AddressInput/AddressInput.tsx:32]: I think just adding `onChange` to the dependency array should work.

File: src/components/ActionsBuilder/SupportedActions/RepMint/RepMintEditor.tsx [hunk 2]
```diff
@@ -76,27 +81,41 @@ export const Mint: React.FC<ActionEditorProps> = ({
...
-            />
-          </ControlRow>
-        </Control>
+        <Controller
+          name="recipient"
+          control={control}
+          render={({ field: { ref, ...field }, fieldState }) => {
+            const { error } = fieldState;
+
+            console.log('here');
```
Comment (rossneilson) [inline src/components/ActionsBuilder/SupportedActions/RepMint/RepMintEditor.tsx:90]: console log here ;)

**Review by rossneilson — APPROVED** [review]

**Pull request merged**